File: backend/Data/Entities/Device.cs
namespace backend.Data.Entities;

public class Device
{
    public int Id { get; set; }
    public string AssetCode { get; set; } = "";
    public string Name { get; set; } = "";
    public string Category { get; set; } = "";
    public string Brand { get; set; } = "";
    public int RoomId { get; set; }
    public int OwnerId { get; set; }
    public string Status { get; set; } = "";
    public string Warranty { get; set; } = "";
    public string Serial { get; set; } = "";
    public string PurchaseDate { get; set; } = "";
    public string UpdatedAt { get; set; } = "";

    // Navigation properties
    public Room? Room { get; set; }
    public AppUser? Owner { get; set; }
}

File: frontend/src/types/device.ts
import type { KpiDto, PointDto } from './common'

export type DeviceStatus = 'Hoạt động' | 'Đang sửa' | 'Bảo trì' | 'Hỏng' | 'Chờ thanh lý'

export interface DeviceItem {
  id: string
  assetCode: string
  name: string
  category: string
  brand: string
  room: string
  owner: string
  status: DeviceStatus
  warranty: string
  serial: string
  purchaseDate: string
  updatedAt: string
}

export type DeviceDto = DeviceItem

export interface DeviceManagementResponse {
  overview: KpiDto[]
  statusBreakdown: PointDto[]
  roomLoad: PointDto[]
  devices: DeviceItem[]
  categories: string[]
}